### PR TITLE
[RW-7854][risk=no] Enable notebook-writer-actions test

### PR DESCRIPTION
**Pull request by aweng98**
Description:
Enable test to add back test coverage. But the test takes >10 min to run and create new workspaces in every run. I broke up the test into two smaller tests and refactored test code. New tests each takes <5 min to run and only create new workspaces when running for the first time on daily basis.

**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have added explanatory comments where the logic is not obvious
- [ ] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

File: e2e/tests/notebook/notebook-create-python.spec.ts [hunk 1]
```diff
@@ -67,13 +67,6 @@ describe('Python Kernel Notebook Test', () => {
       })
     ).toMatch(/success$/);
 
-    expect(
-      await notebook.runCodeCell(3, {
-        codeFile: 'resources/python-code/git-ignore-check.py',
-        markdownWorkaround: true
-      })
-    ).toMatch(/success$/);
-
```
Comment (aweng98) [inline e2e/tests/notebook/notebook-create-python.spec.ts:76]: Moved to ``notebook-writer-clone-workspace.spec.ts`.

File: e2e/tests/notebook/notebook-writer-actions.spec.ts [hunk 2]
```diff
@@ -172,18 +117,48 @@ xdescribe('Workspace WRITER Jupyter notebook action tests', () => {
...
+    await copyModal.copyToAnotherWorkspace(workspaceName, notebookName);
+    await waitWhileLoading(page);
 
-    await newAnalysisPage.deleteResource(copyNotebookName, ResourceCard.Notebook);
+    expect(await copyModal.isLoaded()).toBe(true);
+    modalText = await copyModal.getTextContent();
+    const regex = new RegExp(/Notebook with the same name already exists in the targeted workspace/i);
+    expect(modalText.some((text) => regex.exec(text) !== null)).toBe(true);
+    // Close Copy modal
+    await copyModal.clickButton(LinkText.Close, { waitForClose: true });
```
Comment (aweng98) [inline e2e/tests/notebook/notebook-writer-actions.spec.ts:139]: Add new test coverage: Cannot copy notebook to another workspace if that workspace already has a notebook with same name.

File: e2e/tests/notebook/notebook-writer-actions.spec.ts [hunk 3]
```diff
@@ -172,18 +117,48 @@ xdescribe('Workspace WRITER Jupyter notebook action tests', () => {
...
-    expect(notebookCard).toBeTruthy();
+    await verifyWriterNotebookPermission(page, notebookCopyName);
 
-    // Notebook actions Rename, Duplicate, Delete and Copy to another Workspace actions are available for click.
-    const copyNotebookCardMenu = await notebookCard.getSnowmanMenu();
-    expect(await copyNotebookCardMenu.isOptionDisabled(MenuOption.Rename)).toBe(false);
-    expect(await copyNotebookCardMenu.isOptionDisabled(MenuOption.Duplicate)).toBe(false);
-    expect(await copyNotebookCardMenu.isOptionDisabled(MenuOption.Delete)).toBe(false);
-    expect(await copyNotebookCardMenu.isOptionDisabled(MenuOption.CopyToAnotherWorkspace)).toBe(false);
-    await notebookCard.clickSnowmanIcon(); // close menu
```
Comment (aweng98) [inline e2e/tests/notebook/notebook-writer-actions.spec.ts:185]: Created new function `verifyWriterNotebookPermission()` to remove code duplication.

File: e2e/tests/notebook/notebook-writer-actions.spec.ts [hunk 4]
```diff
@@ -1,120 +1,76 @@
...
         markdownWorkaround: true
       })
     ).toMatch(/success$/);
 
-    await notebook.save();
-    await signOut(page);
+    await notebookPage.save();
   });
 
-  test('WRITER can clone workspace and edit notebook in workspace clone', async () => {
```
Comment (aweng98) [inline e2e/tests/notebook/notebook-writer-actions.spec.ts:52]: Moved to new test `notebook-writer-clone-workspace.ts`.

**Review by chavanr2019 — APPROVED** [review]
LGTM!

**Pull request merged**